### PR TITLE
update font rules, indention, faces and readme

### PR DESCRIPTION
**Pull request by craynot**
Hi!

This merge fixes compitability with latest [tree-sitter-php](https://github.com/tree-sitter/tree-sitter-php), adds some font rules, updates doc and fixes indention

**1.  Fixes compitability with latest [tree-sitter-php](https://github.com/tree-sitter/tree-sitter-php)**
In latest versions node `string_value` has been changed to `string_content`

**2. Add fontifications to base class and `?` symbol in optional types**
`(base_clause (name) @php-class)`
`(optional_type "?" @php-type)`

**3. Add fontification to `$this`**
Looks like `$` symbol is not allowed in capture-name's in queries, so related faces was renamed. To avoid conflicts, where  php-face is already loaded from `php-mode`, those faces was moved to another file `php-ts-face.el`
Also there was a problem with overriding fonts by rules for all others variables and sigils, so this rules was moved to separate feature. If there is a cleaner way to do this, please let me know.

**4. Fix indention of open bracket**
Indention of open bracket in method declaration did not work, it was fixed by adding
`(parent-is "method_declaration") parent-bol 0)`   to `php-ts-mode--indent-rules`

**5. Update README**
Added simple installation & configuration tips.

**6. Fix indention of default expression in switch-case**
Correspondent test was added 

**7. Add face tests**
This is my first experience with Eask, so maybe there is a better way to test font faces.
And maybe it is a good idea to split face test to different `erts` files for each `feature`.  Having one big file or 15 small files, what is better is debatable

**8. Add some font rules**
Mostly it is about qualified_name, for each case test was added

**9. Add `built-in` feature**
To highlight built-in functions, like in `php-mode`

**10. Fix indention of match statement**

### CODE REVIEW

**Comment by piotrkwiecinski** [comment]
Hi @craynot,
thank you for submitting the pull request. Overall it looks good.

As mentioned in the README there are plans to transfer this project to GNU Emacs, so if you haven't assigned a copyright to the FSF yet, please refer to the [Org-mode contribute guide](https://orgmode.org/worg/org-contribute.html#copyright) and send an email to Kenta: tadsan@zonu.me

File: php-ts-mode.el
Comment (piotrkwiecinski) [inline php-ts-mode.el:80]: Would you be able to add a test for this case?
We have very basic suite here: https://github.com/emacs-php/php-ts-mode/blob/master/tests/php-ts-mode-resources/indent.erts

It would be great to increase the coverage while we fix issues.

**Comment by craynot** [comment]
Hi @piotrkwiecinski 
Thank you for the reply!

I have added test for this case. 

Can you clarify about FSF? I have send the [request](https://orgmode.org/request-assign-future.txt)  to [assign@gnu.org](mailto:assign@gnu.org), now waiting for the reply. Or should i have send it to  [tadsan@zonu.me](mailto:tadsan@zonu.me) ?

**Comment by piotrkwiecinski** [comment]
Hi @craynot. 
FSF is going to send you a form to sign. 
Once you sign it, send it back to FSF. 
They'll send you the final copy signed by them.
Please send the final copy to tadsan@zonu.me .

It may take some time but you have to sign it only one time.

**Comment by craynot** [comment]
Thank you @piotrkwiecinski

File: php-ts-face.el
Comment (piotrkwiecinski) [inline php-ts-face.el]: I'm not sure if we need this file. We already have a face for $this here:
https://github.com/emacs-php/php-ts-mode/blob/master/php-face.el#L159-L167

Comment (craynot) [reply php-ts-face.el:33]: If `php-mode` was loaded before, file `php-face.el` from `php-ts-mode.el` is not evaluated

File: php-ts-mode.el
Comment (piotrkwiecinski) [inline php-ts-mode.el]: Echo is already in `php-ts-mode--keywords` https://github.com/emacs-php/php-ts-mode/blob/ef67457ec31462bf78b615d6b47b64f413843d65/php-ts-mode.el#L98C41-L98C47 

If we add `isset` shouldn't we also have `unset` here ?

File: php-ts-face.el
Comment (craynot) [reply php-ts-face.el]: $ symbol is not allowed in capture-name's in queries, so using original faces from `php-mode` is not possible.

Comment (piotrkwiecinski) [reply php-ts-face.el]: Maybe we should just rename them. What do you think @zonuexe?

File: php-ts-mode.el
Comment (craynot) [reply php-ts-mode.el]: `echo` was removed from `php-ts-mode--built-in-functions`
`unset` is parsed differenly by treesitter, it is a `unset_statement`, not simple `function_call_expression`, so individual font rule was added

File: php-ts-face.el
Comment (zonuexe) [reply php-ts-face.el]: @craynot @piotrkwiecinski Sorry for the late reply.

I have [renamed them](https://github.com/emacs-php/php-ts-mode/pull/69) to face names without the `$`, so please run `git rebase master`. The `php-ts-face.el` files should no longer be needed.

Comment (craynot) [reply php-ts-face.el]: @zonuexe thank you! 
`php-ts-face.el` was removed

File: php-ts-mode.el
Comment (zonuexe) [inline php-ts-mode.el:270]: These keywords are [language constructs](https://www.php.net/manual/en/reserved.keywords.php) that are distinct from [built-in functions](https://www.php.net/manual/en/functions.internal.php) and can be misleading to users. Could you please merge these into `keyword`?

Comment (craynot) [reply php-ts-mode.el:270]: @zonuexe  `built-in` feature was removed, rules was merged into `keyword`

Comment (piotrkwiecinski) [reply php-ts-mode.el:270]: As we treat `die`, `empty` and `isset` as keywords then we should just move them to `php-ts-mode--keywords`

Comment (craynot) [reply php-ts-mode.el:270]: @piotrkwiecinski i tried it first thing, but adding any of those values into `php-ts-mode--keywords` breaks [this rule](https://github.com/emacs-php/php-ts-mode/blob/6c0214eb1a323cfc9cedb5e187c4ed20da763e9b/php-ts-mode.el#L243)
You can check this out by evaluating
```
(treesit-query-validate 'php `(["break"] @php-keyword))
```
Query is valid

```
(treesit-query-validate 'php `(["die"] @php-keyword))
```
Raises node error

File: php-ts-mode.el
Comment (piotrkwiecinski) [reply php-ts-mode.el]: I see now. `die`, `empty` and `isset` are represented as:
```
 (expression_statement
  (conditional_expression
   condition: 
    (function_call_expression function: (name)
     arguments: 
      (arguments (
       (argument
        (variable_name $ (name)))
       )))
```
Makes sense.

**Comment by piotrkwiecinski** [comment]
Nice rainbow :smile: 

php-ts-rainbow

**Review by piotrkwiecinski — APPROVED** [review]

**Comment by piotrkwiecinski** [comment]
Great work @craynot. As soon as we have signed FSF doc it's fine to merge @zonuexe.

**Review by zonuexe — APPROVED** [review]
I've received a copy of the FSF copyright assignment from @craynot so we can merge this PR. Thanks for all your help!

**Pull request merged**
